aws_lambda_powertools/utilities/parser/envelopes/base.py

```diff
@@ -11,14 +11,14 @@ class BaseEnvelope(ABC):
     """ABC implementation for creating a supported Envelope"""
 
     @staticmethod
-    def _parse(data: Union[Dict[str, Any], str], model: BaseModel) -> Any:
+    def _parse(data: Union[Dict[str, Any], str], model: Model) -> Union[Model, None]:
```

Comment:
I guess `Model` argument is `Type[Model]`. these parse methods need a defined Model class. It does not need to be instanced.
I found a near example in the official repo.
https://github.com/samuelcolvin/pydantic/blob/29e3877a448e9cb37e058c0a1e10e47a61676b18/pydantic/tools.py#L33

Reply:
Also, I recommend to run mypy checking types.

Reply:
Address that and other findings with mypy - Thanks for the suggestion :) 

I'll create a maintenance issue to discuss with the core maintainers on the feasibility of adding it as part of the code checks - Reason we didn't add earlier was to lower the bar for contributions.